support-workers/src/main/scala/com/gu/emailservices/PaperFieldsGenerator.scala

```diff
@@ -1,51 +1,70 @@
 package com.gu.emailservices
 
-import com.gu.i18n.Currency
-import com.gu.support.promotions.Promotion
+import com.gu.i18n.{Country, Currency}
+import com.gu.support.catalog.ProductRatePlanId
+import com.gu.support.promotions.{PromoCode, Promotion, PromotionService}
 import com.gu.support.workers._
-import com.gu.support.workers.states.PaymentMethodWithSchedule
+import com.gu.support.workers.states.PurchaseInfo
 import org.joda.time.LocalDate
 
-object PaperFieldsGenerator {
+import scala.concurrent.{ExecutionContext, Future}
+
+class PaperFieldsGenerator(
+  promotionService: PromotionService,
+  getMandate: String => Future[Option[String]],
+) {
+
+  def getAppliedPromotion(
+    maybePromoCode: Option[PromoCode],
+    country: Country,
+    productRatePlanId: ProductRatePlanId
+  ): Option[Promotion] =
+    for {
+      promoCode <- maybePromoCode
+      promotionWithCode <- promotionService.findPromotion(promoCode).toOption
+      validPromotion <- promotionService.validatePromotion(promotionWithCode, country, productRatePlanId, isRenewal = false).toOption
+    } yield validPromotion.promotion
```

Comment:
this probably shouldn't really be here in the long term as promos also apply to digital subs.